sklearn/utils/estimator_checks.py

```diff
@@ -418,55 +438,198 @@ def check_transformer_pickle(name, Transformer):
     assert_array_almost_equal(pickled_X_pred, X_pred)
 
 
-def check_estimators_partial_fit_n_features(name, Alg):
-    # check if number of features changes between calls to partial_fit.
-    if not hasattr(Alg, 'partial_fit'):
-        return
+@ignore_warnings
+def check_partial_fit_n_features(name, Estimator):
+    """Check if ValueError is raised if number of features changes between
+    multiple partial_fit calls.
+    """
+
     X, y = make_blobs(n_samples=50, random_state=1)
     X -= X.min()
-    with warnings.catch_warnings(record=True):
-        alg = Alg()
-    set_fast_parameters(alg)
-    if isinstance(alg, ClassifierMixin):
-        classes = np.unique(y)
-        alg.partial_fit(X, y, classes=classes)
-    else:
-        alg.partial_fit(X, y)
 
-    assert_raises(ValueError, alg.partial_fit, X[:, :-1], y)
+    estimator = Estimator()
+
+    set_random_state(estimator)
+    set_fast_parameters(estimator)
+
+    classes = np.unique(y)
+    _partial_fit(estimator, X, y, classes)
+    assert_raises(ValueError, _partial_fit, estimator, X[:, :-1], y)
+
+
+@ignore_warnings
+def check_partial_fit_general(name, Estimator):
+    """General tests for partial_fit method"""
+
+    X, y = make_blobs(n_samples=50, random_state=0)
+    test_X, test_y = make_blobs(n_samples=50, random_state=1)
 
+    if name in ("MultinomialNB",):
+        X -= X.min()
+        test_X -= test_X.min()
+
+    estimator = Estimator()
+
+    set_random_state(estimator)
+    set_fast_parameters(estimator)
+
+    cloned = clone(estimator)
+
+    msg = ("partial_fit of the clone of estimator %s returned a different"
+           " model" % name)
+
+    classes = np.unique(y)
+    ret_val = _partial_fit(estimator, X, y, classes=classes)
+
+    # Check if the partial_fit of cloned returns the same model
+    assert_same_model(test_X, estimator,
+                      _partial_fit(cloned, X, y, classes), msg)
+
+    # Check if returns self
+    msg = "partial_fit of estimator %s did not return self" % name
+    assert ret_val is estimator, msg
+
+
+@ignore_warnings
+def check_partial_fit_reset_when_fit(name, Estimator):
+    """Check if the estimators get reset when fit is applied after a
+    partial fit.
+    """
```

Comment:
@jnothman From your recent comment, I assume changing this to a comment is not necessary?